lptree.pas

```diff
@@ -5652,11 +5763,35 @@ function TLapeTree_For.CompileBody(var Offset: Integer): TResVar;
   IncDec: TLapeTree_InternalMethod;
   tmpBody: TLapeTree_Base;
   tmpContinueStatements: TLapeFlowStatementList;
+  limVar,tmpVar: TResVar;
 begin
   Assert((FCondition <> nil) and (FCondition is TLapeTree_Operator));
   Assert((TLapeTree_Operator(FCondition).Right <> nil) and (TLapeTree_Operator(FCondition).Right is TLapeTree_ResVar));
 
   FStartBodyOffset := FCompiler.Emitter.CheckOffset(Offset);
+  
+  if walkIn then
+  begin
+    FLimit.CompileToTempVar(Offset, limVar);
```

Comment:
Compiling FLimit here might be problematic. For example, what if this has a side-effect such as a function call?